coalib/results/Diff.py

```diff
@@ -70,6 +72,64 @@ def from_string_arrays(cls, file_array_1, file_array_2, rename=False):
         return result
 
     @classmethod
+    def from_unified_diff(cls, unified_diff, original_file):
+        """
+        Creates a ``Diff`` object from given unified diff.
+
+        If this ``Diff`` is applied to the original file, the corrected
+        file will be created.
+
+        :param unified_diff:  Unified diff string.
+        :param original_file: The contents of the original file
+                              (line-splitted).
+        """
+        diff_array = unified_diff.splitlines()
+
+        corrected_file = []
+
+        # index in original_file upto which corrected_file has been generated
+        pos = 0
+
+        patch_set = unidiff.PatchSet(diff_array)
+        for patched_file in patch_set:
+            for hunk in patched_file:
+                changes_array = []
+                source_array = list(hunk.source_lines())
+                target_array = list(hunk.target_lines())
+                source_len, target_len = len(source_array), len(target_array)
+                source_idx, target_idx = 0, 0
+                k = 0
+                while(k < source_len + target_len):
+                    if (source_idx < source_len and
+                            source_array[source_idx].is_context):
+                        changes_array.append(source_array[source_idx])
+                        source_idx += 1
+                        target_idx += 1
+                        k += 1
+                    elif (source_idx < source_len and
+                          source_array[source_idx].is_removed):
+                        changes_array.append(source_array[source_idx])
+                        source_idx += 1
+                    if (target_idx < target_len and
+                            target_array[target_idx].is_added):
+                        changes_array.append(target_array[target_idx])
+                        target_idx += 1
+                    k += 1
+                for line in changes_array:
+                    if line.is_context:
+                        # find occurrence of this line in original file
+                        new_pos = original_file.index(str(line)[1:], pos)
+                        # between pos and new_pos everything is unchanged
+                        for n in range(pos, new_pos+1):
+                            corrected_file.append(original_file[n])
+                        pos = new_pos+1
+                    elif line.is_added:
+                        corrected_file.append(str(line)[1:])
+                    else:
+                        pos += 1
```

Comment:
Actually that's the only way I could think of for generating complete "modified file" from partial unified-diff, and as per previous disussion, it is necessary to generate complete files to create `Diff` object correctly.